src/main/java/pl/drivewheelsdeals/app/controller/ProductController.java

```diff
@@ -31,7 +41,116 @@ public Page<Product> listProducts(Integer page, Integer size) {
         return productService.getAllProducts(pageRequest);
     }
 
+    @PostMapping("/product/add")
+    public ProductOperationResponse addProduct(@Valid @RequestBody ProductOperationRequest request) throws BadRequestException {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        if (auth == null || !auth.isAuthenticated()) {
+            throw new BadRequestException("User is not authenticated");
+        }
+        User user = (User) auth.getPrincipal();
+        if (user == null) {
+            throw new BadRequestException("User not found in the authentication context");
+        }
+        if (!userService.isAdministrator(user)) {
+            throw new BadRequestException("User is not an admin");
+        }
+
+        if(request.product_type.equals("Car")){
+            Car product = new Car();
+            product.setBrand(request.brand);
+            product.setModel(request.model);
+            product.setYear(request.year);
+            product.setPrice(request.price);
+
+            Product added = productService.createProduct(product);
+
+            return new ProductOperationResponse("success -> created product id: " + added.getId());
```

Comment:
imho it would be better to return created product instead of message that it was created